jakarta-eclipse/org.eclipse.lsp4jakarta.core/src/org/jakarta/codeAction/proposal/ModifyModifiersProposal.java

```diff
@@ -0,0 +1,177 @@
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation, Matthew Shocrylas - initial API and implementation
+ *******************************************************************************/
+
+package org.jakarta.codeAction.proposal;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.dom.AST;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.AnonymousClassDeclaration;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.core.dom.FieldDeclaration;
+import org.eclipse.jdt.core.dom.IBinding;
+import org.eclipse.jdt.core.dom.MethodDeclaration;
+import org.eclipse.jdt.core.dom.Modifier;
+import org.eclipse.jdt.core.dom.TypeDeclaration;
+import org.eclipse.jdt.core.dom.VariableDeclarationFragment;
+import org.eclipse.jdt.core.dom.rewrite.ASTRewrite;
+import org.eclipse.jdt.core.dom.rewrite.ImportRewrite;
+import org.eclipse.jdt.core.dom.rewrite.ListRewrite;
+import org.eclipse.jdt.core.dom.rewrite.ImportRewrite.ImportRewriteContext;
+import org.eclipse.jdt.internal.core.manipulation.dom.ASTResolving;
+import org.eclipse.jdt.internal.corext.codemanipulation.ContextSensitiveImportRewriteContext;
+import org.eclipse.lsp4j.CodeActionKind;
+
+/**
+ * Code action proposal for changing the visibility modifier of a method, field, or type.
+ * Used by JAX-RS ResourceMethodQuickFix.
+ * 
+ * @author  Matthew Shocrylas
+ * @author  Leslie Dawson
+ * @see     CodeActionHandler
+ * @see     ResourceMethodQuickFix
+ * @see     PersistenceEntityQuickFix
+ * 
+ */
+public class ModifyModifiersProposal extends ChangeCorrectionProposal {
+
+    private final CompilationUnit invocationNode;
+    private final IBinding binding;
+    private final ASTNode coveredNode;
+    
+    // list of modifiers to add
+    private final List<String> modifiersToAdd;
+
+    // list of modifiers (if they exist) to remove
+    private final List<String> modifiersToRemove;
+    
+    /**
+     * Constructor for ChangeVisibilityProposal. 
+     * 
+     * @param visibility    a valid visibility modifier which will replace the method's current one.
+     * 
+     */
+    public ModifyModifiersProposal(String label, ICompilationUnit targetCU, CompilationUnit invocationNode,
+            IBinding binding, int relevance, ASTNode coveredNode, String visibility) {
+        super(label, CodeActionKind.QuickFix, targetCU, null, relevance);
+        this.invocationNode = invocationNode;
+        this.binding = binding;
+        this.coveredNode = coveredNode;
+        this.modifiersToAdd = Arrays.asList(visibility);
+        this.modifiersToRemove = new ArrayList<>();
+    }
+
+    public ModifyModifiersProposal(String label, ICompilationUnit targetCU, CompilationUnit invocationNode,
```

Comment:
Let's add a comment to this method to clarify that it accepts both a list of modifiers to add and a list of modifiers to remove. 

Add a similar comment to the below method to clarify that it only accepts a list of modifiers to add

Reply:
56c1b47